app/src/main/java/com/example/admin/personallibrarycatalogue/AddBookActivityFragment.java

```diff
@@ -0,0 +1,191 @@
+                             Bundle savedInstanceState) {
+
+
+        final View rootView = inflater.inflate(R.layout.fragment_add_book, container, false);
+        final LibraryDatabaseHelper helper = new LibraryDatabaseHelper(getActivity());
+
+        Bundle extras = getActivity().getIntent().getExtras();
+
+        if (extras != null) {
+            isEditMode_ = true;
+            id_ = getArguments().getInt("id");
+
+            fillAllFields(rootView, helper, id_);
+        }
+
+        imageView_ = (ImageView) rootView.findViewById(R.id.cover_image_view);
+
+        // Actions for change cover button
+        Button setImageButton = (Button) rootView.findViewById(R.id.set_image_button);
+
+        setImageButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent photoPickerIntent = new Intent(Intent.ACTION_PICK);
+                photoPickerIntent.setType("image/*");
+                startActivityForResult(photoPickerIntent, SELECT_PHOTO);
+            }
+        });
+
+        // Actions for ok - button (Checking fields, insert into database, start new activity)
+        Button okButton = (Button) rootView.findViewById(R.id.apply_button);
+
+        okButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                EditText authorEditText = (EditText) rootView.findViewById(R.id.author_edit_text);
+                String author = authorEditText.getText().toString();
+
+                EditText titleEditText = (EditText) rootView.findViewById(R.id.title_edit_text);
+                String title = titleEditText.getText().toString();
+
+                EditText descriptionEditText = (EditText) rootView.findViewById(R.id.description_edit_text);
+                String description = descriptionEditText.getText().toString();
+
+                ImageView coverView = (ImageView) rootView.findViewById(R.id.cover_image_view);
+                byte[] cover = Util.getBytesFromDrawable(coverView.getDrawable());
+
+                if (author.matches("")) {
+                    Toast.makeText(getActivity(), PLEASE_ENTER_THE_AUTHOR_FIELD, Toast.LENGTH_SHORT).show();
+                    return;
+                }
+
+                if (title.matches("")) {
+                    Toast.makeText(getActivity(), PLEASE_ENTER_THE_TITLE_FIELD, Toast.LENGTH_SHORT).show();
+                    return;
+                }
+
+                Book book = new Book(id_,title, author, description, cover);
+
+                // in case user edit information just update data
+                if (id_ != null) {
+                    helper.updateBook(id_, book);
+                } else {
+                    helper.addBook(book);
+                }
+
+                Intent intent = new Intent(getActivity(), BooksListActivity.class);
+                startActivity(intent);
```

Comment:
Bug 
1. Add a book
2. Press back button
You will be switched to add book again.